test/test_type.jl

```diff
@@ -1299,14 +1503,53 @@ isdir(outdir) && rm(outdir, recursive = true)
 
             x = SVector(zero(RealT))
             t = zero(RealT)
-            u = gradients = SVector(one(RealT))
+            u = u_inner = flux_inner = normal = gradients = SVector(one(RealT))
             orientation = 1
 
+            operator_gradient = Trixi.Gradient()
+            operator_divergence = Trixi.Divergence()
+
             @test eltype(@inferred flux(u, gradients, orientation, equations_parabolic)) ==
                   RealT
 
-            # TODO: BC tests for BoundaryConditionDirichlet
-            # TODO: BC tests for BoundaryConditionNeumann
+            # For BC tests
+            function initial_condition_convergence_test(x, t,
+                                                        equation::LaplaceDiffusion1D)
+                RealT_local = eltype(x)
+                x_trans = x[1] - equation.diffusivity * t
+
+                c = 1
+                A = 0.5f0
+                L = 2
+                f = 1.0f0 / L
+                omega = 2 * convert(RealT_local, pi) * f
+                scalar = c + A * sin(omega * sum(x_trans))
+                return SVector(scalar)
+            end
```

Comment:
BC tests here essentially test the type stability of function given here - I cannot find a good example elsewhere so I defined one with no practical meaning.

If you want one with practical meaning, please request a change with yours.